lib/gui/models/settings.js

```diff
@@ -72,3 +55,46 @@ exports.get = (key) => {
 exports.getAll = () => {
   return Store.getState().get('settings').toJS();
 };
+
+exports.setAll = (object) => {
+  return settings.writeAll(settings.getConfigurationFilePath(), object).then(() => {
+    Store.dispatch({
+      type: Store.Actions.SET_SETTINGS,
+      data: object
+    });
+  });
+};
+
+/**
+ * @summary Set a setting value
+ * @function
+ * @public
+ *
+ * @param {String} key - setting key
+ * @param {*} value - setting value
+ * @returns {Promise}
+ *
+ * @example
+ * settings.set('unmountOnSuccess', true).then(() => {
+ *   console.log('Done!');
+ * });
+ */
+exports.set = (key, value) => {
+  if (_.isNil(key)) {
+    return Bluebird.reject(errors.createError({
+      title: 'Missing setting key'
+    }));
+  }
+
+  if (!_.isString(key)) {
+    return Bluebird.reject(errors.createError({
+      title: `Invalid setting key: ${key}`
+    }));
+  }
+
+  const newSettings = _.merge(exports.getAll(), _.fromPairs([
+    [ key, value ]
+  ]));
+
+  return exports.setAll(newSettings);
+};
```

Comment:
In the spirit of CRUD, does there need to be / should there be a way of _deleting_ settings too?